pkg/agent/ipassigner/linkdetector/link_detector_linux_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkdetector
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netlink/nl"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/sys/unix"
+	"k8s.io/utils/set"
+
+	netlinktesting "antrea.io/antrea/pkg/agent/util/netlink/testing"
+)
+
+type linkEventHandler struct {
+	watchLinkNames     []string
+	receivedEvents     []string
+	lock               sync.Mutex
+	expectedLinkEvents []string
+}
+
+func (l *linkEventHandler) onLinkEvent(linkName string) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	l.receivedEvents = append(l.receivedEvents, linkName)
+}
```

Comment:
I would recommend adding a `getReceivedEvents` method, so that the test cases don't need to worry about having to lock the mutex correctly. `getReceivedEvents` should lock the mutex and return a copy of the slice.